File: algorithms/1431.kids-with-the-greatest-number-of-candies/solution.rs
impl Solution {
    pub fn kids_with_candies(candies: Vec<i32>, extra_candies: i32) -> Vec<bool> {
        let max_candies = candies.iter().max().unwrap();
        let mut res: Vec<bool> = Vec::with_capacity(candies.len());

        candies.iter()
            .map(|candy| {
                candy + extra_candies >= *max_candies
            })
            .collect()
    }
}

File: algorithms/1431.kids-with-the-greatest-number-of-candies/Solution.java
class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        int maxCandies = 0;
        List<Boolean> res = new ArrayList<Boolean>(candies.length);

        for (int candy : candies) {
            if (candy > maxCandies) {
                maxCandies = candy;
            }
        }

        for (int candy : candies) {
            res.add(candy + extraCandies >= maxCandies);
        }

        return res;
    }
}
